src/odemis/driver/tescan.py

```diff
@@ -1195,6 +1170,25 @@ def _applyAutoContrastBrightness(self, future):
             with self.parent._acq_progress_lock:
                 self._device_handler.DtAutoSignal(self._channel)
 
+    def _initialize_detectors(self):
+        detectors_raw = self._device_handler.DtEnumDetectors()
+        if not detectors_raw:
+            raise ValueError("No detectors found on the connected device")
+        # Tescan has different formats for different systems.
+        # The det/name format is used on the Solaris and Amber.
+        detectors_tuple = re.findall(r'det\.\d+\.name=(.*?)\ndet\.\d+\.detector=(\d+)', detectors_raw)
+        if not detectors_tuple:
+            raise ValueError("No detectors found on the connected device. Unexpected format.")
+        self._detector_map = {det[0]: int(det[1]) for det in detectors_tuple}
+
+    def _set_detector_type(self, det_name):
+        if det_name not in self._detector_map:
+            logging.warning(f"Detector {det_name} not found among available detectors: "
+                             f"{self._detector_names}")
```

Comment:
This is a VA setter. So the standard VA code already check that det_name is within the choices. This code should never be triggered. You can drop it.